util/syspolicy/rsop/rsop.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package rsop facilitates [source.Store] registration via [RegisterStore]
+// and provides access to the resultant policy merged from all registered sources
+// via [PolicyFor].
+package rsop
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"sync"
+
+	"tailscale.com/syncs"
+	"tailscale.com/util/slicesx"
+	"tailscale.com/util/syspolicy/internal"
+	"tailscale.com/util/syspolicy/setting"
+	"tailscale.com/util/syspolicy/source"
+)
+
+var (
+	policyMu          sync.Mutex
+	policySources     []*source.Source // all registered policy sources
+	resultantPolicies []*Policy        // all active (non-closed) resultant policies returned by [PolicyFor]
```

Comment:
I don't really love the term "resultant policy" because it's kind of a mouthful but I also don't have an obviously better suggesiton.

Maybe "merged policy" or "effective policy"?

Reply:
The term "resultant policy" is Microsoft-speak for essentially the same concept of policy settings that ultimately apply to a user or device. We absolutely don't have to use it; it just sounded natural and familiar to me.

But that aside, I think effective policy is a better name that might also be more familiar to xooglers.
I've renamed "resultant policy" to "effective policy" here and in other places for consistency.